energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs

```diff
@@ -6,7 +6,7 @@ elrond_wasm::derive_imports!();
 pub mod events;
```

Comment:
This module should NOT care about farm token nonces, amounts, anything. This is made to be as generic as possible. It's just a module that splits rewards, per week. We should not pollute this module just because we need something very specific for the farm boosted yields.

Instead, we should have a trait (similar to the one from farm), with more or less the following methods:
- fn calculate_rewards_for_week(&sc, &user, week, &energy);
- fn collect_rewards_for_week(&sc, week);
- fn get_claim_progress(&sc, &user) -> ClaimProgress;
- fn set_claim_progress(&sc, &user, &ClaimProgress);

The handling of farm tokens and such should fall into the wrapper that would implement this trait. Easiest would probably be to have a simple module that holds input payments as a static variable, and provides methods for initializing and accessing those. Then you'd have access to them in a fairly efficient way inside your trait's implementation.

Reply:
I pretty much agree with this.